libindy/tests/metrics.rs

```diff
@@ -16,50 +16,57 @@ mod collect {
     use serde_json::Value;
 
     #[test]
-    fn collect_metrics_contains_thread_pool_and_wallet_service_statistics() {
+    fn collect_metrics_contains_wallet_service_statistics() {
         let result_metrics = metrics::collect_metrics().unwrap();
         let metrics_map = serde_json::from_str::<HashMap<String, Value>>(&result_metrics).unwrap();
 
-        assert!(metrics_map.contains_key("threadpool_threads_count"));
         assert!(metrics_map.contains_key("wallet_count"));
 
-        let threadpool_threads_count = metrics_map
-            .get("threadpool_threads_count")
-            .unwrap()
-            .as_array()
-            .unwrap();
         let wallet_count = metrics_map
             .get("wallet_count")
             .unwrap()
             .as_array()
             .unwrap();
 
-        let expected_threadpool_threads_count = [
-            json!({"tags":{"label":"threadpool_active_count"},"value":0}),
-            json!({"tags":{"label":"threadpool_queued_count"},"value":0}),
-            json!({"tags":{"label":"threadpool_max_count"},"value":4}),
-            json!({"tags":{"label":"threadpool_panic_count"},"value":0}),
-        ];
-
         let expected_wallet_count = [
             json!({"tags":{"label":"opened_wallets_count"},"value":0}),
             json!({"tags":{"label":"opened_wallet_ids_count"},"value":0}),
             json!({"tags":{"label":"pending_for_import_wallets_count"},"value":0}),
             json!({"tags":{"label":"pending_for_open_wallets_count"},"value":0}),
         ];
 
-        for command in &expected_threadpool_threads_count {
-            assert!(threadpool_threads_count.contains(&command));
-        }
+        assert!(wallet_count.contains(&expected_wallet_count[0]));
```

Comment:
I think removing level of indirection and writing just `assert!(wallet_count.contains( json!({"tags":{"label":"opened"},"value":0}) );` would be much more readable :) Even better - you could restructure your tests into following separate test cases:
* check generic schema (that metrics are a dictionary of arrays of dictionaries, and those inner dictionaries always have `value` and `tags` fields)
* check that you have all expected top-level metrics (`wallet_count`, `threadpool_thread_count`, `commands_count`, `commands_duration_ms`, `commands_duration_ms_bucket`, etc)
* for each top-level metric check that you have test that it has all expected submetrics with corresponding tags (and I don't think you really need to check values at this level)

Also you it looks like there is a tautology - you already have metric called `wallet_count`, so I suggest to reduce labels just to `opened`, `opened_ids`, `pending_for_import`, `pending_for_open`.